TestsFoundation/TestsFoundation/Extensions/XCUIElementExtensions.swift

```diff
@@ -19,50 +19,241 @@
+                result = label.hasPrefix(expected!)
+            }
+            if result { break } else { sleep(gracePeriod) }
+        }
+        return self
+    }
+
+    @discardableResult
+    func actionUntilElementCondition(action: ElementAction = .tap,
+                                     element: XCUIElement? = nil,
+                                     condition: ElementCondition,
+                                     expected: String? = nil,
+                                     timeout: TimeInterval = defaultTimeout,
+                                     gracePeriod: UInt32 = defaultGracePeriod) -> Bool {
+        tacticalSleep()
+        let deadline = Date().addingTimeInterval(timeout)
+        while Date() < deadline {
+            let actualElement = element ?? self
+            var result = false
+
+            switch condition {
+            case .vanish:
+                result = actualElement.isVanished
+            case .visible:
+                result = actualElement.isVisible
+            case .value:
+                result = actualElement.hasValue(value: expected!)
+            case .label:
+                result = actualElement.hasLabel(label: expected!)
+            case .enabled:
+                result = actualElement.exists && actualElement.isEnabled
+            case .selected:
+                result = actualElement.exists && actualElement.isSelected
+            case .unselected:
+                result = !actualElement.isSelected
+            case .hittable:
+                result = actualElement.isHittable
+            case .labelContaining:
+                result = label.contains(expected!)
+            case .labelHasPrefix:
+                result = label.hasPrefix(expected!)
+            }
+            if result { return true } else {
+                switch action {
+                case .tap: tap()
+                case .swipeUp: app.swipeUp()
+                case .swipeDown: app.swipeDown()
+                case .swipeLeft: app.swipeLeft()
+                case .swipeRight: app.swipeRight()
+                case .showKeyboard: CoreUITestCase.currentTestCase?.send(.showKeyboard, ignoreErrors: true)
+                case .hideKeyboard: CoreUITestCase.currentTestCase?.send(.hideKeyboard, ignoreErrors: true)
+                case .pullToRefresh: app.pullToRefresh()
+                }
+
+                sleep(gracePeriod)
+            }
+        }
+        return false
+    }
+
+    @discardableResult
+    func writeText(text: String) -> XCUIElement {
+        hit()
+        let keyboard = app.find(type: .keyboard)
+        keyboard.actionUntilElementCondition(action: .showKeyboard, condition: .visible)
+        waitUntil(condition: .visible)
+        typeText(text)
+        keyboard.actionUntilElementCondition(action: .hideKeyboard, condition: .vanish)
+        return self
+    }
+
+    @discardableResult
+    func pasteText(text: String, file: StaticString = #file, line: UInt = #line) -> XCUIElement {
```

Comment:
I think the `file` and the `line` params could be removed.

Reply:
Done.